ls1intum/Artemis
# `Programming exercises`: Fix missing gradle feedback on Jenkins

### Checklist
#### General
#### Server
#### Changes affecting Programming Exercises
- [x] I tested **all** changes and their related features with **all** corresponding user types on Test Server 11 (LocalVC).

### Motivation and Context
Fixes #6434

Bamboo duplicates Gradle Feedback and therefore Artemis cuts it in half. However, this is not the case for Jenkis setup, here Feedback currently gets lost.

### Description
I first tried to cut the feedback directly in the bamboo plugin, but I was not able to detect the exercise type (Maven vs Gradle) there. Now we check if we are on a Bamboo setup and only cut the feedback if this is the case.

In order to use the `ProfileService`, I move the code from a static method in `ResultRepository` into its own `ProgrammingExerciseFeedbackCreationService`.  This is where most of the changed lines in this PR come from, the only real change is in the additional check.

### Steps for Testing
Do the following on all 3 supported CI setups (Bamboo, Jenkins, LocalCI):

1. Create a Gradle PE with a test case that produces multiline feedback, e.g.:

```
@Test
 void longFeedback() {
     fail("""
         a
         b
         c
         d
         """);
 }
```

2. Submit a solution as a student
3. Check that the long feedback is shown correctly.

### Review Progress

#### Performance Review
#### Code Reviv
#### Manual Tests

### Test Coverage
changes covered by additional tests.
